searchcore/src/vespa/searchcore/proton/attribute/attributemanager.cpp

```diff
@@ -581,6 +585,17 @@ AttributeManager::asyncForEachAttribute(std::shared_ptr<IAttributeFunctor> func)
     }
 }
 
+void
+AttributeManager::asyncForAttribute(const vespalib::string &name, std::shared_ptr<IAttributeFunctor> func) const {
+    AttributeMap::const_iterator itr = _attributes.find(name);
+    if (itr == _attributes.end() || itr->second.isExtra() || !func) {
+        return;
+    }
+    AttributeVector::SP attrsp = itr->second.getAttribute();
+    _attributeFieldWriter.execute(_attributeFieldWriter.getExecutorId(attrsp->getNamePrefix()),
```

Comment:
using and moving 'attrsp' in the same statement is not good

Reply:
Fixed